airbyte-ci/connectors/regression-testing/regression_testing/main.py

```diff
@@ -0,0 +1,163 @@
+import asyncio
+import logging
+import sys
+from typing import Dict, Optional
+
+import click
+import dagger
+from airbyte_protocol.models import ConfiguredAirbyteCatalog
+
+from .backends import BaseBackend, FileBackend
+from .connector_runner import ConnectorRunner, SecretDict
+from .utils import get_connector, get_connector_config, get_state
+
+logging.basicConfig(level=logging.INFO)
+
+logger = logging.getLogger(__name__)
+
+
+async def _main(
```

Comment:
I'd love to separate connector image retrieval (test bootstrapping) from actual command execution.
I suggest creating a `ConnnectorUnderTest` dataclass which could have the following attribute:
```python
@dataclass
class ConnectorUnderTest:
   technical_name: str, 
   version: semver.Version # or str 
   container: dagger.Container
```
We could add other attribute for reporting if needed, but the key here is the `container` attribute, whose retrieval happens before instantiation.

We can then call `dispatch` on a `ConnectorUnderTest` instance and  keep the metadata attribute at hand during the whole execution flow.

Reply:
👍 I've added the `ConnectorUnderTest` dataclass and added a TODO for handling building of images based on context.